x-pack/test/functional/apps/dataset_quality/dataset_quality_privileges.ts

```diff
@@ -0,0 +1,203 @@
+    });
+
+    describe('User cannot read any logs-*', () => {
+      before(async () => {
+        await createDatasetQualityUserWithRole(security, 'dataset_quality_no_read', []);
+        await PageObjects.security.forceLogout();
+        await PageObjects.security.login(
+          'dataset_quality_no_read',
+          'dataset_quality_no_read-password',
+          {
+            expectSpaceSelector: false,
+          }
+        );
+
+        await PageObjects.datasetQuality.navigateTo();
+      });
+
+      after(async () => {
+        await deleteDatasetQualityUserWithRole(security, 'dataset_quality_no_read');
+      });
+
+      it('shows empty state as user cannot read any dataset', async () => {
+        await testSubjects.existOrFail(
+          PageObjects.datasetQuality.testSubjectSelectors.datasetQualityNoPrivilegesEmptyState
+        );
+      });
+    });
+
+    describe('User can read logs-*', () => {
+      before(async () => {
+        await createDatasetQualityUserWithRole(security, 'dataset_quality_limited_user', [
+          { names: ['logs-*'], privileges: ['read', 'view_index_metadata'] },
+          { names: ['logs-synth*'], privileges: ['read'] }, // No monitor privilege
+          { names: ['logs-apache*'], privileges: ['monitor'] },
+        ]);
+        await PageObjects.security.forceLogout();
+        await PageObjects.security.login(
+          'dataset_quality_limited_user',
+          'dataset_quality_limited_user-password',
+          {
+            expectSpaceSelector: false,
+          }
+        );
+
+        await PageObjects.datasetQuality.navigateTo();
+      });
+
+      after(async () => {
+        await deleteDatasetQualityUserWithRole(security, 'dataset_quality_limited_user');
+      });
+
+      it('Active and Estimated data are not available due to underprivileged user', async () => {
+        await testSubjects.existOrFail(
+          `${PageObjects.datasetQuality.testSubjectSelectors.datasetQualityInsufficientPrivileges}-${PageObjects.datasetQuality.texts.activeDatasets}`
+        );
+        await testSubjects.existOrFail(
+          `${PageObjects.datasetQuality.testSubjectSelectors.datasetQualityInsufficientPrivileges}-${PageObjects.datasetQuality.texts.estimatedData}`
+        );
+      });
+
+      it('"Show inactive datasets" is hidden when lastActivity is not available', async () => {
+        await find.waitForDeletedByCssSelector(
+          PageObjects.datasetQuality.selectors.showInactiveDatasetsNamesSwitch
+        );
+      });
+
+      it('does not show size and last activity columns for underprivileged data stream', async () => {
+        const cols = await PageObjects.datasetQuality.getDatasetTableHeaderTexts();
+
+        expect(cols).to.not.contain('Size');
+        expect(cols).to.not.contain('Last Activity');
+      });
+
+      it('shows underprivileged warning when size and last activity cannot be accessed for some data streams', async () => {
+        await synthtrace.index(
```

Comment:
We should clean this data ingest as this would impact other tests.
The only clean we have now is at the top level after block, but if more tests are added in the future in this block or parallel block, they will have issues with this data. 

Adding a clean in the after block which is inside this describe could also be a workaround

```
 after(async () => {
     await deleteDatasetQualityUserWithRole(security, 'dataset_quality_limited_user');
 });
```